codalab/worker/docker_utils.py

```diff
@@ -252,3 +254,17 @@ def check_finished(container):
             failure_msg = 'Memory limit exceeded.'
         return (True, exitcode, failure_msg)
     return (False, None, None)
+
+
+@wrap_exception('Unable to check Docker container running time')
+def get_container_running_time(container):
+    # Identical to the docker inspect container command.
+    container_stats = apiclient.inspect_container(container.id)
+    start_time = container_stats['State']['StartedAt']
+    end_time = (
+        container_stats['State']['FinishedAt']
+        if container_stats['State']['Status'] == 'exited'
+        else datetime.datetime.now().replace(tzinfo=tz.tzutc())
```

Comment:
Can you explain the timezone logic here?

Reply:
This is because the DateTime received from docker rest API is using the UTC timezone. I just want to match the current time with what we received from the docker rest API so that the calculation of the total time used is correct.

Reply:
For example, this is the timestamp received from rest api: `"StartedAt": "2019-09-29T03:00:57.558381851Z"`. It's in ISO 8601 format and `Z` stands for UTC timezone.

Reply:
What about using `datetime.utcnow()`?  Also, seems like `end_time` can either be a string or a `datetime` object, which is non-ideal.

Reply:
I tried that earlier today, but got the following:
```
>>> now = datetime.datetime.utcnow()
>>> now.isoformat()
'2019-11-19T19:45:56.941517'
>>> parser.isoparse(now)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/dateutil/parser/isoparser.py", line 37, in func
    return f(self, str_in, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/dateutil/parser/isoparser.py", line 134, in isoparse
    components, pos = self._parse_isodate(dt_str)
  File "/usr/local/lib/python3.6/site-packages/dateutil/parser/isoparser.py", line 208, in _parse_isodate
    return self._parse_isodate_common(dt_str)
  File "/usr/local/lib/python3.6/site-packages/dateutil/parser/isoparser.py", line 213, in _parse_isodate_common
    len_str = len(dt_str)
TypeError: object of type 'datetime.datetime' has no len()
```
But let me double check.